## Github Repository for
# Global diversity and biogeography of the <i> Zostera marina </i> mycobiome 
## by Cassie Ettinger


<i>This work is not yet published.</i>


### To cite this work or code
Ettinger C.L. 2020. casett/Global_ZM_fungi_amplicons v1.0. DOI: https://doi.org/10.5281/zenodo.4116550.

Ettinger C.L., Vann L.E. and Eisen J.A., 2020. Global diversity and biogeography of the Zostera marina mycobiome. bioRxiv. DOI: https://doi.org/10.1101/2020.10.29.361022

### Abstract
Seagrasses are marine flowering plants that provide critical ecosystem services in coastal environments worldwide. Marine fungi are often overlooked in microbiome and seagrass studies, despite terrestrial fungi having critical functional roles as decomposers, pathogens or endophytes in global ecosystems. Here we characterize the distribution of fungi associated with the seagrass, Zostera marina, using leaves, roots, and rhizosphere sediment from 16 locations across its full biogeographic range. Using high throughput sequencing of the ribosomal internal transcribed spacer (ITS) region and 18S ribosomal RNA gene, we first measured fungal community composition and diversity, then we tested hypotheses of neutral community assembly theory and the degree to which deviations suggested amplicon sequence variants (ASVs) were plant-selected or dispersal-limited, and finally we identified a core mycobiome and investigated the global distribution of differentially abundant ASVs. Our results show that the fungal community is significantly different between sites and follows a weak, but significant pattern of distance decay. Generally, there was evidence for both deterministic and stochastic factors contributing to community assembly of the mycobiome. The Z. marina core leaf and root mycobiomes are dominated by unclassified Sordariomycetes spp., unclassified Chytridiomycota lineages (including Lobulomycetaceae spp.), unclassified Capnodiales spp. and Saccharomyces sp. A few ASVs (e.g. Lobulomyces sp.) appear restricted to one or a handful of locations (e.g. possibly due to local adaptation, deterministic dispersal limitation or seasonal bloom events), while others (e.g. Saccharomyces sp.) are more ubiquitous across all locations suggesting a true global distribution and possible plant-selection. Fungal guilds associated with Z. marina were only weakly identified (10.12% of ITS region and 3.4% 18S rRNA gene ASV guild assignments were considered highly probable) including wood saprotrophs, ectomycorrhizal fungi, endophytic fungi and plant pathogens. Our results are similar to those found for other seagrass species. It is clear from the many unclassified fungal ASVs and fungal functional guilds, that our knowledge of marine fungi is still rudimentary. Further studies characterizing seagrass-associated fungi are needed to understand the roles of these microorganisms generally and when associated with seagrasses.

### Funding
This work was supported in part by a community sequencing proposal from the U.S. Department of Energy Joint Genome Institute, "Population and evolutionary genomics of Zostera marina and its microbiome, and genome diagnostics of other seagrass species‚Äù (proposal ID# 503251). The work conducted by the U.S. Department of Energy Joint Genome Institute was further supported by the Office of Science of the U.S. Department of Energy under Contract No. DE-AC02-05CH11231. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.

